### PR TITLE
Update FLIP endpoints

### PR DESCRIPTION
**Pull request by mehranrashid**
Successful run of training

image

### CODE REVIEW

**Review by haroon-younis99 — DISMISSED** [review]

File: flip-app/custom/flip.py [hunk 1]
```diff
@@ -5,25 +5,30 @@
...
         """Calls the data-access-api in FLIP to return a dataframe.
 
         Returns:
             DataFrame: pandas dataframe
         """
 
-    def get_data(self, project_id: str, net_id: str) -> Path:
-        """Calls the imaging-service in FLIP to return a filepath that contains images and labels downloaded from XNAT.
+    def get_by_accession_number(self, project_id: str, accession_id: str) -> Path:
+        """Calls the imaging-service to return a filepath that contains images downloaded from XNAT based
```
Comment (robg-answer) [inline flip-app/custom/flip.py:23]: I'm not sure this comment is relevant.
I don't think the AI Centre will have no/any knowledge of what the Imaging Service is.

File: flip-app/custom/flip.py [hunk 2]
```diff
@@ -5,25 +5,30 @@
...
+           on the accession number
 
         Returns:
-            Path: path of data
+            Path: path to data
```
Comment (robg-answer) [inline flip-app/custom/flip.py:27]: This needs updating

File: flip-app/custom/flip.py [hunk 3]
```diff
@@ -5,25 +5,30 @@
...
     def get_dataframe(self, project_id: str, query: str) -> DataFrame:
         """Calls the data-access-api in FLIP to return a dataframe.
 
         Returns:
             DataFrame: pandas dataframe
         """
 
-    def get_data(self, project_id: str, net_id: str) -> Path:
-        """Calls the imaging-service in FLIP to return a filepath that contains images and labels downloaded from XNAT.
+    def get_by_accession_number(self, project_id: str, accession_id: str) -> Path:
```
Comment (robg-answer) [inline flip-app/custom/flip.py:22]: This no longer returns a path.
Although admittedly we might not know what it is at this stage.

Comment (mehranrashid) [reply flip-app/custom/flip.py:22]: Imaging service returns the path https://gitlab.com/AnswerDigital/londonai/flip/services/imaging-service/-/merge_requests/31/diffs

File: flip-app/custom/flip.py (hunk 2 above)
Comment (mehranrashid) [reply flip-app/custom/flip.py:27]: Imaging service returns a path Imaging service returns the path https://gitlab.com/AnswerDigital/londonai/flip/services/imaging-service/-/merge_requests/31/diffs

File: flip-app/custom/flip.py (hunk 1 above)
Comment (mehranrashid) [reply flip-app/custom/flip.py:23]: True. Comments about the imaging service were already there, thought I'd keep it consistent. I'll change it to just `a service`

File: flip-app/custom/flip.py [hunk 4]
```diff
@@ -5,25 +5,30 @@
...
         Returns:
-            Path: path of data
+            Path: path to data
+        """
+
+    def add_resource(self, project_id: str, accession_id: str, scan_id: str, resource_id: str, files: List[str]):
+        """Calls a service to upload image(s) to XNAT based on the accession number, scan ID, and resource ID
         """
 
     def update_status(self, model_id: str, new_model_status: ModelStatus):
```
Comment (robg-answer) [inline flip-app/custom/flip.py:34]: This needs moving out of this module.
Potentially in the actual FLIP module itself.

Comment (yorkeanswer) [reply flip-app/custom/flip.py:34]: Isn't this the flip module?

File: flip-app/custom/flip.py (hunk 1 above)
Comment (yorkeanswer) [reply flip-app/custom/flip.py]: Could we refer to them as `FLIP service(s)`?

File: flip-app/custom/flip.py (hunk 4 above)
Comment (robg-answer) [reply flip-app/custom/flip.py:34]: As per discussions were going to have to leave this in

File: README.md [hunk 5]
```diff
@@ -70,15 +70,32 @@ shutdown server
...
-   already passed in as parameters to the trainer to be used.
-
-- `get_data(self, project_id: str, net_id: str) -> Path`
-   This method is for internal use only, and is not be called by the trainer.
-
-- `update_status(self, model_id: str, new_model_status: ModelStatus)`
-   This method is for internal use only, and is not be called by the trainer.
+  This retrieves data in the form of a Dataframe containing, at the minimum, accession IDs.
+  The method takes in the project ID and the project query as parameters. These values are
+  already passed in as parameters to the trainer to be used.
```
Comment (yorkeanswer) [inline README.md:75]: Didn't we say we were removing the `projectId`?

File: README.md [hunk 6]
```diff
@@ -70,15 +70,32 @@ shutdown server
...
+
+- `get_by_accession_number(self, project_id: str, accession_id: str) -> Path`
+  This downloads scans and places them in a directory made available for NVFlare to utilise.
+  The method takes in the project ID as a parameter as well as an accession ID, which can be 
+  obtained from `get_dataframe`. It returns the path to where the scans are stored.
+
+- `add_resource(self, project_id: str, accession_id: str, scan_id: str, resource_id: str, files: List[str])`
+  This allows uploading scans to XNAT under the project that the model to. Scans are to be placed
```
Comment (yorkeanswer) [inline README.md:83]:
```suggestion
  This allows uploading scans to XNAT under the project that the model belongs to. Scans are to be placed 
```

**Review by robg-answer — DISMISSED** [review]

File: flip-app/custom/cleanup.py [hunk 7]
```diff
@@ -1,3 +1,5 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
```
Comment (yorkeanswer) [inline flip-app/custom/cleanup.py:2]: This is our file, not NVIDIAs

File: flip-app/custom/pre_training.py [hunk 8]
```diff
@@ -27,24 +27,23 @@
...
-        The DataRetrieval workflow sends a request to each of the participating clients to retrieve the images that it
-            will use for training
+        The InitTraining workflow sends a request to the Cental Hub, stating that training has initiated
 
         Args:
             model_id (str): ID of the model that the training is being performed under.
             min_clients (int, optional): Minimum number of clients. Defaults to 2 for the aggregation to take place with
                 successful results.
-            retrieval_task_name (str, optional): Name of the retrieval task. Defaults to "retrieve_images".
+            task_name (str, optional): Name of the task. Defaults to "init_training".
```
Comment (robg-answer) [inline flip-app/custom/pre_training.py:46]: Might want to add FLIP as an arg to this doc

File: flip-app/custom/flip.py [hunk 9]
```diff
@@ -1,3 +1,5 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
```
Comment (yorkeanswer) [inline flip-app/custom/flip.py:2]: Same, not NVIDIAs

```suggestion

```

File: flip-app/custom/flip_server_event_handler.py [hunk 10]
```diff
@@ -1,3 +1,5 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
```
Comment (yorkeanswer) [inline flip-app/custom/flip_server_event_handler.py:2]: ditto

File: flip-app/custom/utils/utils.py [hunk 11]
```diff
@@ -1,3 +1,5 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
```
Comment (yorkeanswer) [inline flip-app/custom/utils/utils.py:2]: Remove this 👍

File: flip-app/custom/pre_training.py (hunk 8 above)
Comment (mehranrashid) [reply flip-app/custom/pre_training.py:46]: 🦅 👁️

File: README.md (hunk 5 above)
Comment (mehranrashid) [reply README.md:75]: We said that's another ticket?

File: flip-app/custom/cleanup.py (hunk 7 above)
Comment (mehranrashid) [reply flip-app/custom/cleanup.py:2]: My bad, thought it all belonged to them since we extended it

**Review by carlsmith-ad — APPROVED** [review]

**Review by robg-answer — APPROVED** [review]

**Review by robg-answer — APPROVED** [review]

**Review by carlsmith-ad — APPROVED** [review]

**Pull request merged**